File: codes/js/problems/lcof/54.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *   this.val = val;
 *   this.left = this.right = null;
 * }
 */
/**
 * 递归
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
const kthLargest = function(root, k) {
  const res = []
  const dfs = (node) => {
    if (!node) return
    dfs(node.left)
    res.push(node.val)
    dfs(node.right)
  }
  dfs(root)
  return res[res.length - k]
};

File: py3/lcof/54.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def kthLargest(self, root: TreeNode, k: int) -> int:
        stack, vals = [], []
        ptr = root
        while ptr or stack:
            while ptr:
                stack.append(ptr)
                ptr = ptr.left

            ptr = stack.pop()
            vals.append(ptr.val)
            ptr = ptr.right

        return vals[-k]